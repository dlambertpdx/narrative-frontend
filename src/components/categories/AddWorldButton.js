import React from 'react';
import PropTypes from 'prop-types';
import EditFormModalWorld from './EditFormModalWorld';

const AddWorldButton = ({ show, hideModal, showModal }) => {
  return (
    <>
      <EditFormModalWorld show={show} hideModal={hideModal}></EditFormModalWorld>
      <div>
        <button type='button' onClick={showModal}>Add New</button>
      </div>
    </>
  );
};

AddWorldButton.propTypes = {
  show: PropTypes.bool,
  hideModal: PropTypes.func,
  showModal: PropTypes.func
};

export default AddWorldButton;
