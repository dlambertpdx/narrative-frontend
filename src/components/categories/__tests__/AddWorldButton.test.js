import React from 'react';
import { shallow } from 'enzyme';
import AddWorldButton from '../AddWorldButton';

describe('AddWorldButton component', () => {
  it('renders an add button', () => {
    const wrapper = shallow(<AddWorldButton handleAddClick={() => true} />);
    expect(wrapper).toMatchSnapshot();
  });
});