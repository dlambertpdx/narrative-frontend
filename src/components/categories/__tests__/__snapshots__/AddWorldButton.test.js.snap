// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddWorldButton component renders an add button 1`] = `
<Fragment>
  <EditFormModalWorld />
  <div>
    <button
      type="button"
    >
      Add New
    </button>
  </div>
</Fragment>
`;
